llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -612,11 +612,16 @@ static bool isKnownNonZeroFromAssume(const Value *V, const SimplifyQuery &Q) {
     // Warning: This loop can end up being somewhat performance sensitive.
     // We're running this loop for once for each value queried resulting in a
     // runtime of ~O(#assumes * #values).
+    Value *Arg = I->getArgOperand(0);
+    if (match(Arg, m_TruncOrSelf(m_Specific(V))) &&
+        isValidAssumeForContext(I, Q.CxtI, Q.DT)) {
```

Comment:
Looks like we hit another "ephemeral value" issue :(

Reply:
I do not know what issue you are thinking about is it something that is shown in the llvm-opt-benchmark?

Reply:
Yeah. The test diff shows it only removes some (unused?) assumptions.

Reply:
I do not know if there is that many assumptions like this right now but was looking at folding some icmp eq X, C to trunc X to i1 when X is known to be 0 or 1 and got lots of regressions due to this.